config/nrfconnect/chip-module/generate_factory_data.cmake

```diff
@@ -0,0 +1,204 @@
+function(nrfconnect_create_factory_data_hex_file factory_data_target script_path output_path)
+
+if(NOT CONFIG_CHIP_FACTORY_DATA_PARTITION_ADDRESS OR NOT CONFIG_CHIP_FACTORY_DATA_PARTITION_SIZE)
+message(FATAL_ERROR "Can not generate factory_data hex file. Please ensure that CONFIG_CHIP_FACTORY_DATA_PARTITION_ADDRESS and CONFIG_CHIP_FACTORY_DATA_PARTITION_SIZE were provided")
+endif()
+
+# Pass the argument list via file
+set(cbor_script_args "-i ${output_path}/${factory_data_target}.json\n")
+string(APPEND cbor_script_args "-o ${output_path}/${factory_data_target}\n")
+string(APPEND cbor_script_args "--offset ${CONFIG_CHIP_FACTORY_DATA_PARTITION_ADDRESS}\n")
+string(APPEND cbor_script_args "--size ${CONFIG_CHIP_FACTORY_DATA_PARTITION_SIZE}\n")
+# Pass the argument list via file
+file(GENERATE
+    OUTPUT ${output_path}/${factory_data_target}_cbor.args
+    CONTENT ${cbor_script_args})
+
+# execute second script to create a hex file containing factory data in cbor format
+separate_arguments(separated_cbor_script_args NATIVE_COMMAND ${cbor_script_args})
+add_custom_command(TARGET ${factory_data_target} POST_BUILD
+    COMMAND ${Python3_EXECUTABLE} ${script_path} ${separated_cbor_script_args}
+    COMMENT "Generating factory data HEX file..."
+    )
+
+endfunction()
+
+# Generate factory data partition using given args
+#
+# 
+# During generation process a some file will be created in zephyr's build directory:
+# - merged_fd.hex a file containing firmware and factory data merged to single file
+# - factory_data.hex a file containing only a factory data partition including proper offset
+#
+# [Args]:
+#   target_name - a name for taget to generate factory_data
+function(nrfconnect_generate_factory_data target_name)
+
+find_package(Python REQUIRED)
+
+# CHIP_ROOT must be provided as a reference set all localizations of scripts
+if(NOT CHIP_ROOT)
+message(FATAL_ERROR "CHIP_ROOT variable is not set, please add it to CMakeLists.txt file")
+endif()
+
+# Localize all scripts needed to generate factory data partition
+set(FACTORY_DATA_SCRIPT_PATH ${CHIP_ROOT}/scripts/tools/nrfconnect/generate_nrfconnect_chip_factory_data.py)
+set(GENERATE_CBOR_SCRIPT_PATH ${CHIP_ROOT}/scripts/tools/nrfconnect/nrfconnect_generate_partition.py)
+SET(MERGE_HEX_SCRIPT_PATH ${CHIP_ROOT}/config/nrfconnect/chip-module/merge_factory_data.py)
+set(FACTORY_DATA_SCHEMA_PATH ${CHIP_ROOT}/scripts/tools/nrfconnect/nrfconnect_factory_data.schema)
+set(OUTPUT_FILE_PATH ${APPLICATION_BINARY_DIR}/zephyr)
+
+# create a JSON file with all factory data
+nrfconnect_create_factory_data_json(${target_name} ${FACTORY_DATA_SCRIPT_PATH} ${FACTORY_DATA_SCHEMA_PATH} ${OUTPUT_FILE_PATH})
+
+# create a .hex file with factory data in CBOR format based on the JSON file created previously 
+nrfconnect_create_factory_data_hex_file(${target_name} ${GENERATE_CBOR_SCRIPT_PATH} ${OUTPUT_FILE_PATH})
+
+# merge firmware .hex with factory_data.hex to output merged_fd.hex
```

Comment:
Is this merged_fd.hex final file that we should program to the board? If so I think there is a problem, that `west flash` will not be able to do that...

Reply:
I will try to merge it to merge.hex file using partition_manager.cmake mechanism. I mean that we can use pm_static.yaml file and partition_manager.cmake to add factory_data.hex file to a list of all .hex files merged after building.